hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HReplicationServer.java

```diff
@@ -78,27 +104,44 @@
    */
   private ServerName serverName;
 
-  protected final Configuration conf;
+  private final Configuration conf;
 
-  private ReplicationSinkService replicationSinkService;
+  // zookeeper connection and watcher
+  private final ZKWatcher zooKeeper;
 
-  final int msgInterval;
-  // A sleeper that sleeps for msgInterval.
-  protected final Sleeper sleeper;
+  private final UUID clusterId;
 
-  // zookeeper connection and watcher
-  protected final ZKWatcher zooKeeper;
+  private HFileSystem walFs;
+  private Path walRootDir;
+
+  /**
+   * ChoreService used to schedule tasks that we want to run periodically
+   */
+  private ChoreService choreService;
 
   /**
    * The asynchronous cluster connection to be shared by services.
    */
-  protected AsyncClusterConnection asyncClusterConnection;
+  private AsyncClusterConnection asyncClusterConnection;
 
   private UserProvider userProvider;
 
-  protected final ReplicationServerRpcServices rpcServices;
+  private final ReplicationServerRpcServices rpcServices;
```

Comment:
We have to do this? Just because we do it for HMaster and HRegionServer, it doesn't make it a good pattern. This is an Interface to pass instead of the impl?

Reply:
Maybe it is to mock?

Reply:
I am ok with this. Decouple a RpcService can make the code more clearly.